src/platforms/wayland/displayclient.cpp

```diff
@@ -123,9 +118,84 @@ static EGLint const ctxattribs[] =
     };
 }
 
+mgw::DisplayClient::Output::Output(
+    wl_output* output,
+    DisplayClient* owner,
+    std::function<void(Output const&)> on_constructed,
+    std::function<void(Output const&)> on_change) :
+    output{output},
+    owner{owner},
+    surface{wl_compositor_create_surface(owner->compositor)},
+    on_done{[this, on_constructed = std::move(on_constructed), on_change=std::move(on_change)]
+        (Output const& o) mutable { on_constructed(o), on_done = std::move(on_change); }}
```

Comment:
We're now passing the thing as both `on_constructed` and `on_change`. I think we can simplify this further.

Reply:
Since we're always finishing construction by the end of the `DisplayClient` constructor, we don't need `on_constructed()` at all anymore.